gen/enum.go

```diff
@@ -53,6 +53,29 @@ func (e *enumGenerator) Reader(g Generator, spec *compile.EnumSpec) (string, err
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
 
+func (e *enumGenerator) Decoder(g Generator, spec *compile.EnumSpec) (string, error) {
```

Comment:
Be consistent with calling the function "Decode" like you've done with the fieldGenerator.